tests/traction_multiple_model.prm

```diff
@@ -0,0 +1,152 @@
+set Start time                             = 0
+set Use years in output instead of seconds = true
+
+subsection Boundary temperature model
+  set List of model names = initial temperature
+
+  subsection Initial temperature
+    set Minimal temperature = 293
+    set Maximal temperature = 1600
+  end
+end
+
+subsection Geometry model
+  set Model name = box
+  subsection Box
+    set X extent  = 3000000
+    set Y extent  = 660000
+    set X repetitions = 9
+    set Y repetitions = 2
+  end
+
+end
+
+
+subsection Gravity model
+  set Model name = vertical
+  subsection Vertical
+    set Magnitude = 10.0
+  end
+end
+
+
+subsection Initial temperature model
+  set Model name = adiabatic
+  subsection Adiabatic
+    set Age top boundary layer      = 1e8
+    set Age bottom boundary layer   = 0.0
+    set Amplitude                   = 250
+    set Radius                      = 400000
+  end
+end
+
+subsection Material model
+
+  set Model name = diffusion dislocation
+  subsection Diffusion dislocation
+    set Densities                                 = 3330
+    set Maximum viscosity                         = 1e26
+    set Minimum viscosity                         = 1e17
+    set Heat capacity                             = 1200
+    set Thermal expansivities                     = 3.3e-5
+    set Activation energies for diffusion creep   = 375000
+    set Activation volumes for diffusion creep    = 6e-6
+    set Grain size exponents for diffusion creep  = 4
+    set Prefactors for diffusion creep            = 1.5e-16
+    set Activation energies for dislocation creep = 480000
+    set Prefactors for dislocation creep          = 2e-15
+    set Activation volumes for dislocation creep  = 8e-6
+    set Maximum strain rate ratio iterations      = 10000
+  end
+end
+
+
+subsection Mesh refinement
+  set Initial adaptive refinement              = 0
+  set Initial global refinement                = 4
+  set Time steps between mesh refinement       = 0
+end
+
+# The parameters below this comment were created by the update script
+# as replacement for the old 'Model settings' subsection. They can be
+# safely merged with any existing subsections with the same name.
+
+subsection Boundary temperature model
+  set Fixed temperature boundary indicators   = 3
+end
+
+subsection Boundary traction model
+  set Prescribed traction boundary indicators = 0:ascii data, 0:function
+end
+
+subsection Boundary traction model
+   subsection Function
+    set Variable names      = x,y
+
+    # The following line is the stress applied to the side boundaries (in Pa)
+    set Function constants  = stress=2000e6
+    set Function expression = if (y>600e3 ,-stress,0); 0
+  end
+end
+subsection Boundary velocity model
+  set Tangential velocity boundary indicators = 1,2,3
+end
+
+subsection Boundary traction model
+  subsection Ascii data model
+    set Data directory =$ASPECT_SOURCE_DIR/data/boundary-traction/ascii-data/test/
+  end
```

Comment:
You have three occurrences of `subsection Boundary traction model` here. Please merge all of them into one subsection (one `subsection Boundary traction model` .... `end` block).